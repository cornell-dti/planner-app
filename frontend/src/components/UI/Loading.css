.LoadingTextWrapper {
  display: flex;
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
}

.LoadingText {
  margin: auto;
  text-align: center;
  font-size: 3em;
  font-weight: 300;
}
