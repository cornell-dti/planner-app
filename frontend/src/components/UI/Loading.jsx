// @flow strict

import React from 'react';
import styles from './Loading.module.css';

export default () => (
  <div className={styles.LoadingTextWrapper}>
    <div className={styles.LoadingText}>Loading...</div>
  </div>
);
