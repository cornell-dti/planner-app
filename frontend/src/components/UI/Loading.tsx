import React, { ReactElement } from 'react';
import styles from './Loading.module.css';

export default (): ReactElement => (
  <div className={styles.LoadingTextWrapper}>
    <div className={styles.LoadingText}>Loading...</div>
  </div>
);
