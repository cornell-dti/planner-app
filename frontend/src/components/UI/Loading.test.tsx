import React from 'react';
import { renderIntoDocument } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import Loading from './Loading';

it('Loading can render', () => {
  renderIntoDocument(<Loading />);
});

it('Loading matches snapshot.', () => {
  const tree = renderer.create(<Loading />).toJSON();
  expect(tree).toMatchSnapshot();
});
