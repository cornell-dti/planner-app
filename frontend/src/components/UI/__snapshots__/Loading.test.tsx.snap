// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading matches snapshot. 1`] = `
<div
  className="LoadingTextWrapper"
>
  <div
    className="LoadingText"
  >
    Loading...
  </div>
</div>
`;
